src/inmanta/data/__init__.py

```diff
@@ -2937,6 +2937,75 @@ async def get_deleted_resources(cls, environment, current_version, current_resou
 
         return should_purge
 
+    @classmethod
+    async def get_resource_details(cls, env: uuid.UUID, resource_id: m.ResourceIdStr) -> Optional[m.ResourceDetails]:
+        status_subquery = """
+        (CASE WHEN
+            (SELECT resource.model < MAX(configurationmodel.version)
+            FROM configurationmodel
+            WHERE configurationmodel.released=TRUE
+            AND environment = $1)
+            THEN 'orphaned'
+        ELSE resource.status::text END)
+        as status
+        """
+
+        query = f"""
```

Comment:
I would also add some comments in the query, to indicate what the different subqueries do.